docker-compose.yml

```diff
@@ -14,30 +14,28 @@ services:
 
   web:
     extends: base
-    command: web -b 0.0.0.0:8000
+    command: web -b 0.0.0.0:8000 --reload
     ports:
       - "8080:8000"
-    links:
-      - mysql
-      - redis
     environment:
-      DEBUG: 1
+      - DEBUG=1
+      - ALLOWED_HOSTS=127.0.0.1,localhost,web
 
   worker:
     extends: base
-    command: worker -l info --queues localhost,celery
-    links:
-      - prometheus
-      - mysql
-      - redis
+    command: worker -l info --queues localhost,celery,prometheus,alertmanager
     volumes:
       - ./docker:/etc/prometheus
 
   prometheus:
     image: prom/prometheus
-    links:
-      - alertmanager
-      - blackbox
+    user: "root"
```

Comment:
In general, I think it's probably best to run as a non-root user

Reply:
I agree with you that it should be runned as a non-root user.

The original issue is that prometheus couldn't create the required files&folders.
```
prometheus_1    | level=error ts=2020-06-05T13:30:24.569Z caller=query_logger.go:87 component=activeQueryTracker msg="Error opening query log file" file=/prometheus/queries.active err="open /prometheus/queries.active: permission denied"
prometheus_1    | panic: Unable to create mmap-ed active query log
```
This is due to the fact that by default the folders were created with the following user:groups

``` bash
drwxr-xr-x 6 systemd-coredump root       4096 Jun  5 09:30 mysql_data
drwxr-xr-x 2 root             root       4096 Jun  5 09:30 prom_data
```
It looks like prometheus uses the nobody user https://github.com/prometheus/prometheus/blob/master/Dockerfile#L19

What do you think about using the user nobody as prometheus does?

The folder creation could then be done by an entrypoint script.

Reply:
Using the same prometheus user (nobody) should be fine.